src/Extsload.sol

```diff
@@ -17,42 +17,46 @@ abstract contract Extsload is IExtsload {
     /// @inheritdoc IExtsload
     function extsload(bytes32 startSlot, uint256 nSlots) external view returns (bytes memory) {
         assembly ("memory-safe") {
-            // The abi offset of dynamic array in the returndata is 32.
-            mstore(0, 0x20)
+            let memptr := mload(0x40)
```

Comment:
I thought we didn't have to care about memory safety here, because the view function is external

Reply:
here I mean, because the function is external, it enters a new call context and it exits right after, so we should be able to disregard the memory conventions

Reply:
Yeah I had a long conversation with our auditors about this. This is the only issue that was raised by _all 3_ audit firms.
Yes it is correct that the frame ends after this external call, but the code does not follow the conventions of memory safely. While it is not currently vulnerable to an attack that any of them can find, it is not future proof - because the assembly is marked as `memory-safe` but it does not follow the rules of memory safety. So if any changes are ever made to how memory is handled, this code might become unsafe due to the fact its been labelled as safe but isnt.